### DISCUSSION TITLE
pull oci charts preemptive to template command

### DISCUSSION THREAD

**Discussion by travisghansen** [helmfile/helmfile#727] [Questions]
I maintain https://github.com/travisghansen/argo-cd-helmfile and an issue has come up https://github.com/travisghansen/argo-cd-helmfile/issues/36 where during the template process the args passed to helm are invalid.

`argo-cd` plugins operate in 2 phases:

- `init`
- `generate`

The approach I took was to have everything prepped (repos added/updated etc) during the `init` phase to ensure that *only* actual templating occurs during the `generate` phase.

I don't really have any experience with `oci` registries but it appears this is handled fundamentally different than the 'old' stuff. Is there a way I can pull the `oci` charts before the `template` sub-command runs to ensure no fetching is done? I see `helmfile fetch` is a new sub-command but I cannot tell if there's some way I could tell `template` to use the previously fetched charts.

**Comment by yxxhero** [comment]
@travisghansen could you add some examples or any logs? Thanks very much.

**Reply by travisghansen** [reply]
Output is here: https://github.com/travisghansen/argo-cd-helmfile/issues/36#issuecomment-1453298929

**Comment by yxxhero** [comment]
@travisghansen Thanks for your feedback. but I am sorry that I don't know the logic of argo-cd-helmfile. could you give me more background?

**Reply by travisghansen** [reply]
The log in the linked comment is triggered from `helmfile template` after invoking `helmfile repos`. I would like to ensure that during the template subcommand I have all charts already available so it doesn’t blow up like that.

**Reply by yxxhero** [reply]
@travisghansen  got it. let me help you.

**Reply by travisghansen** [reply]
@yxxhero any further thoughts on this?

**Comment by yxxhero** [comment]
@travisghansen `helmfile repos` just only add chart reports. it doesn't pull the chart. `helmfile template` will prepare the chat before doing `helm template`.

**Reply by travisghansen** [reply]
Yeah, you've stated everything I already know. The original question was about pulling the chart preemptive to the template command. I don't want to pull the charts as part of the template operation.

**Comment by yxxhero** [comment]
@travisghansen maybe you need a flag to disable prepare chart in `helmfile template`. right?

**Reply by travisghansen** [reply]
Would it be possible to revisit this issue?
